File: linked-list/removeloop.cpp
 void removeLoop(Node* head)
    {
        Node *slow = head;
       Node *fast = head;

       //Detecting whether a loop is present or not
       while(slow!=NULL && fast!=NULL && fast->next!=NULL){
           fast=fast->next->next;
            slow=slow->next;
           
       
           if(slow == fast) {
               
              break;
           }
       }
       slow=head;
        
        //edge case
        if(slow == fast) {
            while(fast->next != slow) {
                fast = fast->next;
            }
            
            fast->next = NULL;
            return;
        }

        //Traversing just before the starting node of loop
        while(slow->next != fast->next){
           slow = slow->next;
           fast = fast->next;
        }
    
       //Make the last node point to NULL
       fast->next = NULL;  
    }
};

File: linked-list/removeloop.js
var detectCycle = function(head) {
    let slow= head;
    let fast = head;

    while( fast && fast.next && fast.next.next){
        slow=slow.next;
        fast=fast.next.next;

        if(slow===fast){ //loop is detected.
                slow=head;//slow variable is takn back to head node and fast variable is there where it is .
                while(slow!==fast){ //then again starting from starting head node
                    slow=slow.next;
                    fast=fast.next;
                }
                fast.next=null;
                return slow;
        }
    }
    return null;


};
